backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py

```diff
@@ -459,7 +467,7 @@ def _fetch_response(
         tool: CustomTool,
         doc_path: str,
         doc_name: str,
-        prompts: list[ToolStudioPrompt],
+        prompt: ToolStudioPrompt,
```

Comment:
@chandrasekharan-zipstack Any reason why this is changed from a list to single object?
When a tool_id is passed to this API, [this](https://github.com/Zipstack/unstract/blob/6cf77545b9faef06ba0f089a3f3ae4cb54644fb3/backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py#L201) method is invoked to fetch the prompt response, which will return a list of prompts mapped to this Id.

Reply:
Answered [here](https://github.com/Zipstack/unstract/pull/329#discussion_r1598042131) - can you provide a link to the code where we fill this list with multiple prompts? What I noticed was only a single instance being added

Reply:
@chandrasekharan-zipstack `fetch_prompt_from_tool` method retruns a list ( cases where tool_id ) is passed and is invoked [here](https://github.com/Zipstack/unstract/blob/6cf77545b9faef06ba0f089a3f3ae4cb54644fb3/backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py#L403). And in [this](https://github.com/Zipstack/unstract/blob/6cf77545b9faef06ba0f089a3f3ae4cb54644fb3/backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py#L420) line, we access it as a list again. We might need to refactor in these places if we decide to remove the list.
( tool_id

Reply:
[Single pass execution](https://github.com/Zipstack/unstract/blob/6cf77545b9faef06ba0f089a3f3ae4cb54644fb3/backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py#L421C1-L427C18) follows a different flow and though we practically call the same API we call different functions later. I haven't refactored this bit since single pass behaves differently and it needs to be run with all prompts in the project - a list is needed. However in case of a single prompt being executed - a list is unnecessary

Reply:
Got it, on checking the single pass flow.